vimeo-networking/src/main/java/com/vimeo/networking/Configuration.java

```diff
@@ -97,28 +98,49 @@ public Cache getCache() {
     }
 
     /**
-     * @deprecated use {@link #saveAccount(VimeoAccount, String)} instead
-     * <p/>
-     * We find no use in storing the password when you can persist the {@link VimeoAccount} across
-     * application sessions.
+     * Persist the {@link VimeoAccount} in the {@link AccountStore}
+     * <p>
+     * Account name is <b>required</b> for authenticated user accounts; for
+     * client credential account saving use {@link #saveNonUserAccount(VimeoAccount)}
+     *
+     * @param account     the account to persist
+     * @param accountName the name of the account.
      */
-    @Deprecated
-    public void saveAccount(VimeoAccount account, String email, String password) {
-        saveAccount(account, email);
+    public void saveAuthenticatedUserAccount(@NotNull VimeoAccount account, @NotNull String accountName) {
+        if (accountStore != null) {
+            accountStore.saveAuthenticatedUserAccount(account, accountName);
+        }
     }
 
-    public void saveAccount(VimeoAccount account, String email) {
+    /**
+     * Persist the {@link VimeoAccount} in the {@link AccountStore} without an account
+     * name. For authenticated users, use {@link #saveAuthenticatedUserAccount(VimeoAccount, String)},
+     * as this is meant for client credential or code grant saving, when an account name is not available.
+     *
+     * @param account the account to persist
+     */
+    public void saveNonUserAccount(@NotNull VimeoAccount account) {
         if (accountStore != null) {
-            accountStore.saveAccount(account, email);
+            accountStore.saveNonUserAccount(account);
```

Comment:
@alfiehanssen this was the argument brought up in the initial review of this PR - that it would be clearer if we had two separate methods for saving an account rather that one with optional parameters. Both you and @howardrigberg thought that was the preferred approach. It was also suggested I rename "saveAccount" so that it is more pointed as to what is being saved. Are you saying that you are against this approach now?